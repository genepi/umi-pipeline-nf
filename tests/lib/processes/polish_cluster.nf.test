nextflow_process {

    name "Test Process POLISH_CLUSTER"
    script "lib/processes/polish_cluster.nf"
    process "POLISH_CLUSTER"

        setup{
            run("CLUSTER"){
                script "$baseDir/lib/processes/cluster.nf"
                process {
                    """
                    input[0] = Channel.of(
                        ["barcode03", "raw", "$baseDir/tests/input/cluster/barcode03/detected_umis.fastq"],
                        ["barcode04", "raw", "$baseDir/tests/input/cluster/barcode04/detected_umis.fastq"],
                        ["barcode05", "raw", "$baseDir/tests/input/cluster/barcode05/detected_umis.fastq"],
                        ["barcode06", "raw", "$baseDir/tests/input/cluster/barcode06/detected_umis.fastq"],
                        ["barcode07", "raw", "$baseDir/tests/input/cluster/barcode07/detected_umis.fastq"]
                    )
                    input[1] = "raw"
                    """
                }
            }
            run("REFORMAT_FILTER_CLUSTER"){
                script "$baseDir/lib/processes/reformat_filter_cluster.nf"
                params{
                    min_reads_per_cluster = 10
                }
                process {
                    """
                    input[0] = CLUSTER.out.cluster_fastas
                    input[1] = "raw"
                    input[2] = file("$baseDir/bin/parse_clusters.py")
                    """
                }
            }
        }


    test("Should run without failures") {
        when {
            process {
                """
                input[0] = REFORMAT_FILTER_CLUSTER.out.smolecule_cluster_fastqs.transpose()
                input[1] = "Consensus"
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
