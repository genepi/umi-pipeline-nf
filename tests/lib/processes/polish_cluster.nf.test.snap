{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    [
                        "barcode03",
                        "raw",
                        "smolecule0sub0_consensus.fastq:md5,85c055b607a0ae893d1e8d2c0fe298af"
                    ],
                    [
                        "barcode03",
                        "raw",
                        "smolecule1sub0_consensus.fastq:md5,0390b95558912435618e9a0569280a8f"
                    ],
                    [
                        "barcode03",
                        "raw",
                        "smolecule2sub0_consensus.fastq:md5,0ef3b301741224bf4aad61b61d6eb7c2"
                    ],
                    [
                        "barcode03",
                        "raw",
                        "smolecule3sub0_consensus.fastq:md5,7f4723552bfdd42432b6fb52d4483f56"
                    ],
                    [
                        "barcode03",
                        "raw",
                        "smolecule4sub0_consensus.fastq:md5,813d3d07a2fe5a8ca343f758831208fd"
                    ]
                ],
                "consensus_fastq": [
                    [
                        "barcode03",
                        "raw",
                        "smolecule0sub0_consensus.fastq:md5,85c055b607a0ae893d1e8d2c0fe298af"
                    ],
                    [
                        "barcode03",
                        "raw",
                        "smolecule1sub0_consensus.fastq:md5,0390b95558912435618e9a0569280a8f"
                    ],
                    [
                        "barcode03",
                        "raw",
                        "smolecule2sub0_consensus.fastq:md5,0ef3b301741224bf4aad61b61d6eb7c2"
                    ],
                    [
                        "barcode03",
                        "raw",
                        "smolecule3sub0_consensus.fastq:md5,7f4723552bfdd42432b6fb52d4483f56"
                    ],
                    [
                        "barcode03",
                        "raw",
                        "smolecule4sub0_consensus.fastq:md5,813d3d07a2fe5a8ca343f758831208fd"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.0"
        },
        "timestamp": "2024-03-15T13:07:01.670879"
    }
}